numpy/core/src/multiarray/ctors.c

```diff
@@ -806,6 +806,9 @@ PyArray_NewFromDescr_int(
         fa->flags |= NPY_ARRAY_F_CONTIGUOUS;
     }
 
+    /* Store the functions in case the global hander is modified */
+    fa->mem_handler = current_allocator;
```

Comment:
Is there a scenario where someone flips the NPY_ARRAY_OWNDATA flag? The only cases I found in the NumPy codebase are around unpickling in `methods.c` where we are now very careful to use the `fa->mem_handler` routines. But in the name of caution you are correct that it would be a safer thing to do.

Bottom line: adopting.

Reply:
Good point, I think I read that comment below recently and assumed this was documented as one way to it.  I would almost be surprised if nobody does it, but I do agree it should be exceedingly rare.  I guess it was not a hassle to change this, though? Otherwise, we could consider documenting it and since it should crash very loudly I am not super worried.